build_mt.py

```diff
@@ -1,599 +1,827 @@
+    if "Progenitors" in fields:
+        merger_tree, progs = simple_load(fns[start:stop], fields=fields)
     else:
-        merger_tree = simple_load(fns[start:stop],fields=fields)
+        merger_tree = simple_load(fns[start:stop], fields=fields)
 
     # if a far redshift, need 2 copies only
     if copies == 2:
         merger_tree0 = merger_tree
         merger_tree1 = merger_tree.copy()
-        merger_tree1['Position'] += np.array([0,0,Lbox])
+        merger_tree1["Position"] += np.array([0, 0, Lbox])
         merger_tree2 = merger_tree.copy()
-        merger_tree2['Position'] += np.array([0,Lbox,0])
-        merger_tree = np.hstack((merger_tree1,merger_tree2))
-        
+        merger_tree2["Position"] += np.array([0, Lbox, 0])
+        merger_tree = np.hstack((merger_tree1, merger_tree2))
+
     # if in intermediate redshift range, need 3 copies
     elif copies == 3:
         merger_tree0 = merger_tree
         merger_tree1 = merger_tree.copy()
-        merger_tree1['Position'] += np.array([0,0,Lbox])
+        merger_tree1["Position"] += np.array([0, 0, Lbox])
         merger_tree2 = merger_tree.copy()
-        merger_tree2['Position'] += np.array([0,Lbox,0])
-        merger_tree = np.hstack((merger_tree0,merger_tree1,merger_tree2))
-        
+        merger_tree2["Position"] += np.array([0, Lbox, 0])
+        merger_tree = np.hstack((merger_tree0, merger_tree1, merger_tree2))
+
     # get number of halos in each slab and number of slabs
     N_halos_slabs, slabs = get_slab_halo(fns)
-    
+
     # load positions in Mpc/h, index of the main progenitors, index of halo
-    pos = merger_tree['Position']
-    main_prog = merger_tree['MainProgenitor']
-    halo_ind = merger_tree['HaloIndex']
+    pos = merger_tree["Position"]
+    main_prog = merger_tree["MainProgenitor"]
+    halo_ind = merger_tree["HaloIndex"]
 
     # compute comoving distance between observer and every halo
-    dimensions = np.array([Lbox,Lbox,Lbox])
-    com_dist = distance(pos,origin)
-    #com_dist = distance_period(pos,origin,dimensions)
-    
+    com_dist = dist(pos, origin)
+    # com_dist = dist(pos,origin,L=Lbox)  # periodic version
+
     # if loading all progenitors
-    if 'Progenitors' in fields:
-        num_progs = merger_tree['NumProgenitors']
+    if "Progenitors" in fields:
+        num_progs = merger_tree["NumProgenitors"]
         # get an array with the starting indices of the progenitors array
-        start_progs = np.zeros(merger_tree.shape,dtype=int)
+        start_progs = np.zeros(merger_tree.shape, dtype=int)
         start_progs[1:] = num_progs.cumsum()[:-1]
-        
-        return com_dist, main_prog, halo_ind, pos, start_progs, num_progs, progs, N_halos_slabs, slabs
+
+        return (
+            com_dist,
+            main_prog,
+            halo_ind,
+            pos,
+            start_progs,
+            num_progs,
+            progs,
+            N_halos_slabs,
+            slabs,
+        )
```

Comment:
I ran Black (Python code formatter) on this file, so some of the choices are a little strange, like writing long tuples this way. Just a matter of taste, I guess.  But I would definitely advocate for returning all this info in a dict.  But that's jumping ahead a bit.